Remove store throttling.

jpountz opened the pull request:
Store throttling has been disabled by default since Lucene added automatic
throttling of merge operations based on the indexing rate.

jpountz commented on core/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java:
```diff
@@ -887,10 +881,6 @@ private void createTestIndex(String name) {
-        assertAcked(client().admin().indices().prepareUpdateSettings(name).setSettings(Settings.builder()
-                        .put(IndexStore.INDEX_STORE_THROTTLE_TYPE_SETTING.getKey(), "all")
-                        .put(IndexStore.INDEX_STORE_THROTTLE_MAX_BYTES_PER_SEC_SETTING.getKey(), between(100, 50000))
-        ));
```
I'm not sure how to migrate this test: does it still have a value with the above lines removed?

s1monw replied:
it's been disabled forever... I am not sure maybe @imotov knows, the test you are talking about is `testChaosSnapshot` right?

mikemccand commented on core/src/main/java/org/elasticsearch/index/shard/IndexShard.java:
```diff
@@ -672,17 +672,7 @@ public CommitStats commitStats() {
-            throttleTimeInMillis = 0;
-        } else {
-            throttled = engine.isThrottled();
-            throttleTimeInMillis = engine.getIndexThrottleTimeInMillis();
```
Hang on: this is index throttling, which still exists and is different from store throttling?

Index throttling is when ES forces incoming indexing to be single threaded because merges have fallen too far behind ...

jpountz replied:
Oh, good point. I'll fix.

jpountz replied:
OK I understand what happened now. Index throttling seemed to be always disabled so I got rid of it. But it was actually not disabled all the time, it was a bug that `InternalEngine#isThrottled` did not override the default behaviour of `Engine#isThrottled`. I opened a PR: https://github.com/elastic/elasticsearch/pull/21592

mikemccand commented:
+1 to remove store level throttling, but (I think?) we still have index level throttling and should keep its stats?

jpountz commented:
Thanks @s1monw and @mikemccand. I rebased in order to have the fix from #21592, added index throttling back and removed the snapshot test that has been disabled for more than 1 year.

mikemccand commented:
LGTM, thanks @jpountz.  Nice to see all the -

clintongormley commented:
@jpountz should this be marked as breaking?  it was still possible to use it in 5.x and there were stats reported in the node stats which have been removed

jpountz commented:
Good call. It should not break many too many users though since this feature was disabled by default.

jpountz commented:
@clintongormley See https://github.com/elastic/elasticsearch/commit/c7fc68809681c4f2ebc4e6b00ca282a01bd98a53
